apps/condo/domains/ticket/components/BaseTicketForm/TicketDeadlineField.tsx

```diff
@@ -1,70 +1,124 @@
     const AutoCompletionMessage = intl.formatMessage({ id: 'ticket.deadline.AutoCompletion' })
+    const TicketWithoutDeadlineMessage = intl.formatMessage({ id: 'pages.condo.ticket.WithoutDeadline' })
 
     const { isSmall } = useLayoutContext()
+    const { ticketSetting } = useTicketSettingContext()
+    const { isPaid, isEmergency, isWarranty } = form.getFieldsValue(['isPaid', 'isEmergency', 'isWarranty'])
+    const dirtyTicketSettingId = useMemo(() => get(ticketSetting, 'id', null), [ticketSetting])
+    const [ticketSettingId, setTicketSettingId] = useState<string | null>(dirtyTicketSettingId)
 
     const isExistedTicket = get(initialValues, 'property')
+    const isFirstRender = useRef<boolean>(true)
 
-    const [isAutoDetectedValue, setIsAutoDetectedValue] = useState<boolean>(!isExistedTicket)
+    const autoAddDays: null | number = useMemo(
+        () => getTicketDefaultDeadline(ticketSetting, isPaid, isEmergency, isWarranty),
+        [isEmergency, isPaid, isWarranty, ticketSetting]
+    )
+
+    const autoDeadlineValue = useMemo(() => {
+        return isNull(autoAddDays) ? autoAddDays : dayjs().add(autoAddDays, 'day')
+    }, [autoAddDays])
 
     const handleTicketDeadlineChange = useCallback(() => {
         setIsAutoDetectedValue(false)
-    }, [])
+    }, [setIsAutoDetectedValue])
 
-    return (
-        <Row align='bottom' gutter={TICKET_DEADLINE_FIELD_ROW_GUTTER} justify='space-between'>
-            <Col span={isSmall ? 24 : 11}>
-                <TicketFormItem
-                    label={CompleteBeforeMessage}
-                    name='deadline'
-                    required
-                    initialValue={INITIAL_DEADLINE_VALUE}
-                    data-cy='ticket__deadline-item'
-                >
-                    <DatePicker
-                        format='DD MMMM YYYY'
-                        onChange={handleTicketDeadlineChange}
-                        disabledDate={isDateDisabled}
-                        style={DATE_PICKER_STYLE}
-                        allowClear={false}
-                    />
-                </TicketFormItem>
-            </Col>
-            {
-                isAutoDetectedValue && (
-                    <Col style={AUTO_DETECTED_DEADLINE_COL_STYLE} span={isSmall ? 24 : 11}>
-                        <Row justify='start' align='middle' style={AUTO_DETECTED_DEADLINE_ROW_STYLE}>
-                            <Col span={24}>
-                                <Typography.Text type='secondary' style={AUTO_COMPLETE_MESSAGE_STYLE}>
-                                    {AutoCompletionMessage}
-                                </Typography.Text>
-                            </Col>
-                        </Row>
+    const autoDetectedLabel = useMemo(() => {
+        if (!isAutoDetectedValue || isNull(autoAddDays)) return null
+
+        const DaysMessage = intl.formatMessage({ id: 'DaysShort' }, { days: autoAddDays })
```

Comment:
mode it at the top

Reply:
above there is no `autoAddDays` variable